packages/hooks-react/src/useOffers.ts

```diff
@@ -1,45 +1,62 @@
 
 const useOffers = () => {
   const checkoutController = getModule(CheckoutController);
-  const svodOfferIds = checkoutController.getSubscriptionOfferIds();
-  const { requestedMediaOffers: mediaOffers } = useCheckoutStore(({ requestedMediaOffers }) => ({ requestedMediaOffers }), shallow);
-  const hasPremierOffers = mediaOffers?.some((offer) => offer.premier);
-  const hasMultipleOfferTypes = !hasPremierOffers && !!mediaOffers?.length && !!svodOfferIds.length;
-  const offerIds: string[] = mergeOfferIds(mediaOffers || [], svodOfferIds);
-
-  const [offerType, setOfferType] = useState<OfferType>(hasPremierOffers || !svodOfferIds.length ? 'tvod' : 'svod');
-  const updateOfferType = useMemo(() => (hasMultipleOfferTypes ? (type: OfferType) => setOfferType(type) : undefined), [hasMultipleOfferTypes]);
-
-  const { data: allOffers, isLoading } = useQuery(['offers', offerIds.join('-')], () => checkoutController.getOffers({ offerIds }));
-
-  // The `offerQueries` variable mutates on each render which prevents the useMemo to work properly.
-  return useMemo(() => {
-    const offers = (allOffers || []).filter((offer: Offer) => (offerType === 'tvod' ? !isSVODOffer(offer) : isSVODOffer(offer)));
-
-    const offersDict = (!isLoading && Object.fromEntries(offers.map((offer: Offer) => [offer.offerId, offer]))) || {};
-    // we need to get the offerIds from the offer responses since it contains different offerIds based on the customers'
-    // location. E.g. if an offer is configured as `S12345678` it becomes `S12345678_US` in the US.
-    const defaultOfferId = (!isLoading && offers[offers.length - 1]?.offerId) || '';
-
-    return {
-      hasMediaOffers: allOffers?.some((offer: Offer) => !isSVODOffer(offer)),
-      isLoading,
-      defaultOfferId,
-      offerType,
-      setOfferType: updateOfferType,
-      offers,
-      offersDict,
-    };
-  }, [allOffers, isLoading, offerType, updateOfferType]);
+  const accountController = getModule(AccountController);
+
+  const { mediaOffers, subscriptionOffers, switchSubscriptionOffers, requestedMediaOffers } = useCheckoutStore(
+    ({ mediaOffers, subscriptionOffers, switchSubscriptionOffers, requestedMediaOffers }) => ({
+      mediaOffers,
+      subscriptionOffers,
+      switchSubscriptionOffers,
+      requestedMediaOffers,
+    }),
+    shallow,
+  );
+
+  const { mutate: initialise, isLoading: isInitialisationLoading } = useMutation<void>({
+    mutationKey: ['initialiseOffers', requestedMediaOffers],
+    mutationFn: checkoutController.initialiseOffers,
+  });
+
+  const chooseOffer = useMutation({
+    mutationKey: ['chooseOffer'],
+    mutationFn: checkoutController.chooseOffer,
+  });
+
+  const switchSubscription = useMutation({
+    mutationKey: ['switchSubscription'],
+    mutationFn: checkoutController.switchSubscription,
+    onSuccess: () => accountController.reloadSubscriptions({ delay: 7500 }), // @todo: Is there a better way to wait?
```

Comment:
Do we wait for Cleeng here?

Reply:
Both InPlayer and Cleeng need some time to process external payments. Even though the mutation succeeds, the subscription endpoint can yield an outdated response.

FYI @royschut has worked on improving this part by introducing a delay and retry mechanism: https://github.com/Videodock/ott-web-app/pull/147 (needs a rebase)

Reply:
The `7500` duration was copied from what [previously was the submit handler](https://github.com/jwplayer/ott-web-app/blob/develop/packages/ui-react/src/containers/AccountModal/forms/ChooseOffer.tsx#L68) handling a subscription switch, indeed waiting for processing time in the backend.

The [coming PR ](https://github.com/Videodock/ott-web-app/pull/147) that @ChristiaanScheermeijer mentions will wait for a shorter time and simply retry a couple of times if it isn't ready.

Reply:
Thank you!